tests/src/rules/no-unused-modules.js

```diff
@@ -163,6 +173,40 @@ ruleTester.run('no-unused-modules', rule, {
   ],
 })
 
+// test for unused exports with `import()`
+ruleTester.run('no-unused-modules', rule, {
+  valid: [
+    test({ options: unusedExportsOptions,
+           code: `
+           export const a = 10
+           export const b = 20
+           export const c = 30
+           const d = 40
+           export default d
+           `,
+           parser: require.resolve('babel-eslint'),
+           filename: testFilePath('./no-unused-modules/exports-for-dynamic-js.js')}),
+  ],
+  invalid: [
+  ],
+})
+typescriptRuleTester.run('no-unused-modules', rule, {
```

Comment:
this test always passes however corresponding ts file is empty, what am I doing wrong?

Reply:
i believe the `filename` doesn't have to exist, it's just to set the context for rules

Reply:
hmm. how does it check if specific file is imported? I thought filename here is provided specifically to check that this code belongs to specificed file and then track imports of such file. 

Which in case of ts does not work and tests always green.